### ISSUE TITLE
Portmaster don`t open.

### ISSUE THREAD

**Issue by sergiogbrox** [cizia64/CrossMix-OS#119]
My portmaster dont load, dont open before the 1.2.1 version.

log:

[?25l[?1c[?25h[?0ccurrent_mode 1280x720
MALI_CreateWindow:0x558e77b140 done.
2024-07-19 22:55:54.693 | DEBUG    | harbourmaster.config:<module>:109 - HM_DEFAULT_TOOLS_DIR:   /mnt/SDCARD/Apps/PortMaster
2024-07-19 22:55:54.695 | DEBUG    | harbourmaster.config:<module>:110 - HM_DEFAULT_PORTS_DIR:   /mnt/SDCARD/Data/ports
2024-07-19 22:55:54.696 | DEBUG    | harbourmaster.config:<module>:111 - HM_DEFAULT_SCRIPTS_DIR: /mnt/SDCARD/Data/ports
[?25l[?1c2024-07-19 22:55:55.606 | INFO     | __main__:__init__:515 - PM: 2024.07.17-1125
2024-07-19 22:55:55.607 | INFO     | __main__:__init__:516 - HM: 2024-02-27
2024-07-19 22:55:55.608 | INFO     | __main__:__init__:517 - SDL DLL: /usr/trimui/lib/libSDL2-2.0.so.0.2600.5,  2.26.5
2024-07-19 22:55:55.609 | INFO     | __main__:__init__:518 - TTF DLL: /usr/trimui/lib/libSDL2_ttf-2.0.so.0.14.1, 2.0.15
2024-07-19 22:55:55.611 | INFO     | __main__:__init__:519 - IMG DLL: /usr/trimui/lib/libSDL2_image-2.0.so.0.2.3, 2.0.5
2024-07-19 22:55:55.612 | INFO     | __main__:__init__:520 - MIX DLL: /usr/trimui/lib/libSDL2_mixer-2.0.so.0.0.1, 2.0.1
2024-07-19 22:55:55.614 | INFO     | __main__:__init__:530 - Opened GameController 0: b'Xbox 360 Controller'
2024-07-19 22:55:55.615 | INFO     | __main__:__init__:531 -  b'030003f05e0400008e02000014010000,Xbox 360 Controller,a:b0,b:b1,back:b6,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b10,righttrigger:a5,rightx:a3,righty:a4,start:b7,x:b2,y:b3,platform:Linux,'
2024-07-19 22:55:55.621 | DEBUG    | harbourmaster.hardware:device_info:510 - DEVICE INFO: {'name': 'TrimUI', 'version': '1.0.4', 'device': 'trimui-smart-pro', 'resolution': (1280, 720), 'analogsticks': 2, 'cpu': 'a133plus', 'capabilities': ['power', 'aarch64', 'restore', '16:9', '1280x720', 'TrimUI', 'trimui-smart-pro', 'analog_0', 'analog_1', 'analog_2', 'hires', 'wide', '1gb'], 'ram': 1024, 'primary_arch': 'aarch64'}
2024-07-19 22:55:55.622 | INFO     | __main__:__init__:592 - {'name': 'TrimUI', 'version': '1.0.4', 'device': 'trimui-smart-pro', 'resolution': (1280, 720), 'analogsticks': 2, 'cpu': 'a133plus', 'capabilities': ['power', 'aarch64', 'restore', '16:9', '1280x720', 'TrimUI', 'trimui-smart-pro', 'analog_0', 'analog_1', 'analog_2', 'hires', 'wide', '1gb', 'en_US'], 'ram': 1024, 'primary_arch': 'aarch64'}
2024-07-19 22:55:56.025 | INFO     | pugtheme:theme_load:127 - Loading theme /mnt/SDCARD/Apps/PortMaster/PortMaster/themes/hiviz/theme.json
2024-07-19 22:55:56.028 | DEBUG    | pugtheme:theme_load:227 - - loading base_data
2024-07-19 22:55:56.029 | DEBUG    | pugtheme:theme_load:252 - - loading elements:
2024-07-19 22:55:56.031 | DEBUG    | pugtheme:theme_load:298 -   - loading section ports_list
2024-07-19 22:55:56.034 | DEBUG    | pugtheme:theme_load:298 -   - loading section port_info
2024-07-19 22:55:56.036 | DEBUG    | pugtheme:theme_load:298 -   - loading section port_info_popup
2024-07-19 22:55:56.038 | DEBUG    | pugtheme:theme_load:298 -   - loading section main_menu
2024-07-19 22:55:56.040 | DEBUG    | pugtheme:theme_load:298 -   - loading section option_menu
2024-07-19 22:55:56.042 | DEBUG    | pugtheme:theme_load:298 -   - loading section themes_list
2024-07-19 22:55:56.044 | DEBUG    | pugtheme:theme_load:298 -   - loading section runtime_list
2024-07-19 22:55:56.046 | DEBUG    | pugtheme:theme_load:298 -   - loading section on_screen_keyboard
2024-07-19 22:55:56.048 | DEBUG    | pugtheme:theme_load:298 -   - loading section filter_list
2024-07-19 22:55:56.049 | DEBUG    | pugtheme:theme_load:298 -   - loading section message_window
2024-07-19 22:55:56.051 | DEBUG    | pugtheme:theme_load:298 -   - loading section featured_ports_list
2024-07-19 22:55:56.053 | DEBUG    | pugtheme:theme_load:298 -   - loading section message_box
2024-07-19 22:55:56.055 | DEBUG    | pugtheme:theme_load:298 -   - loading section blank
2024-07-19 22:55:56.056 | DEBUG    | pugtheme:theme_load:298 -   - loading section selection_list
2024-07-19 22:55:56.058 | DEBUG    | pugtheme:theme_load:298 -   - loading section selection_list_description
2024-07-19 22:55:56.060 | DEBUG    | pugtheme:theme_load:298 -   - loading section selection_list_images
2024-07-19 22:55:56.061 | DEBUG    | pugtheme:theme_load:298 -   - loading section selection_list_description_images
2024-07-19 22:55:56.063 | DEBUG    | pugtheme:theme_load:275 - - loading pallet:
2024-07-19 22:55:56.064 | DEBUG    | pugtheme:theme_load:231 - - loading resources:
2024-07-19 22:55:56.066 | DEBUG    | pugtheme:theme_load:247 -   - loading sound click.wav as click - [OKAY]
2024-07-19 22:55:56.069 | DEBUG    | pugtheme:theme_load:247 -   - loading sound cancel.wav as cancel - [OKAY]
2024-07-19 22:55:56.071 | DEBUG    | pugtheme:theme_load:247 -   - loading sound scroll.wav as scroll - [OKAY]
2024-07-19 22:55:56.073 | DEBUG    | pugtheme:theme_load:247 -   - loading sound slide_up.wav as slide_up - [OKAY]
2024-07-19 22:55:56.076 | DEBUG    | pugtheme:theme_load:247 -   - loading sound slide_down.wav as slide_down - [OKAY]
2024-07-19 22:55:56.108 | DEBUG    | pugtheme:theme_load:241 -   - loading image buttons.png - [OKAY]
2024-07-19 22:55:56.130 | DEBUG    | pugtheme:theme_load:239 -   - loading image no-image.jpg as NO_IMAGE - [OKAY]
2024-07-19 22:55:56.131 | DEBUG    | pugtheme:theme_load:298 -   - loading section disclaimer
2024-07-19 22:55:56.175 | DEBUG    | __main__:push_scene:1257 - PUSH SCENE LAYER messages
2024-07-19 22:55:56.176 | DEBUG    | __main__:push_scene:1260 - SCENE LIST: ['root', 'messages']
2024-07-19 22:55:56.733 | DEBUG    | __main__:pop_scene:1282 - POP SCENE LAYER messages
2024-07-19 22:55:56.735 | DEBUG    | __main__:pop_scene:1285 - SCENE LIST: ['root']
2024-07-19 22:55:56.736 | ERROR    | __main__:<module>:2029 - An error has been caught in function '<module>', process 'MainProcess' (3701), thread 'MainThread' (548373550976):
Traceback (most recent call last):

> File "/mnt/SDCARD/Apps/PortMaster/PortMaster/./pugwash", line 2029, in <module>
    exit(main(sys.argv))
         │    │   └ ['./pugwash', '--debug']
         │    └ <module 'sys' (built-in)>
         └ <function main at 0x7fabb1dc60>
  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/./pugwash", line 2002, in main
    pm.hm = HarbourMaster(config, temp_dir=temp_dir, callback=pm)
    │  │    │             │                │                  └ <__main__.PortMasterGUI object at 0x7fabe99110>
    │  │    │             │                └ PosixPath('/tmp/tmp8wunhkyb')
    │  │    │             └ {'quiet': False, 'no-check': False, 'debug': True, 'no-colour': False, 'force-colour': False, 'no-log': False, 'help': False,...
    │  │    └ <class 'harbourmaster.harbour.HarbourMaster'>
    │  └ None
    └ <__main__.PortMasterGUI object at 0x7fabe99110>

  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/pylibs/harbourmaster/harbour.py", line 151, in __init__
    self.load_ports()
    │    └ <function HarbourMaster.load_ports at 0x7fac1dd300>
    └ <harbourmaster.harbour.HarbourMaster object at 0x7fabb3ef50>

  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/pylibs/harbourmaster/harbour.py", line 621, in load_ports
    port_info = self._load_port_info(port_file)
                │    │               └ PosixPath('/mnt/SDCARD/Data/ports/fallout1/port.json')
                │    └ <function HarbourMaster._load_port_info at 0x7fac1dcea0>
                └ <harbourmaster.harbour.HarbourMaster object at 0x7fabb3ef50>

  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/pylibs/harbourmaster/harbour.py", line 422, in _load_port_info
    port_info = port_info_load(port_file, do_default=True)
                │              └ PosixPath('/mnt/SDCARD/Data/ports/fallout1/port.json')
                └ <function port_info_load at 0x7fac17df80>

  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/pylibs/harbourmaster/info.py", line 52, in port_info_load
    info = json_safe_load(fh)
           │              └ <_io.TextIOWrapper name='/mnt/SDCARD/Data/ports/fallout1/port.json' mode='r' encoding='utf-8'>
           └ <function json_safe_load at 0x7fac617380>

  File "/mnt/SDCARD/Apps/PortMaster/PortMaster/pylibs/harbourmaster/util.py", line 47, in json_safe_load
    return json.load(*args)
           │    │     └ (<_io.TextIOWrapper name='/mnt/SDCARD/Data/ports/fallout1/port.json' mode='r' encoding='utf-8'>,)
           │    └ <function load at 0x7fad3f6d40>
           └ <module 'json' from '/mnt/SDCARD/System/lib/python3.11/json/__init__.py'>

  File "/mnt/SDCARD/System/lib/python3.11/json/__init__.py", line 293, in load
    return loads(fp.read(),
           │     │  └ <method 'read' of '_io.TextIOWrapper' objects>
           │     └ <_io.TextIOWrapper name='/mnt/SDCARD/Data/ports/fallout1/port.json' mode='r' encoding='utf-8'>
           └ <function loads at 0x7fad3f6de0>
  File "<frozen codecs>", line 322, in decode

UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 6: invalid start byte
0: 1
Display size: 1280x720
{'name': 'TrimUI', 'version': '1.0.4', 'device': 'trimui-smart-pro', 'resolution': (1280, 720), 'analogsticks': 2, 'cpu': 'a133plus', 'capabilities': ['power', 'aarch64', 'restore', '16:9', '1280x720', 'TrimUI', 'trimui-smart-pro', 'analog_0', 'analog_1', 'analog_2', 'hires', 'wide', '1gb', 'en_US'], 'ram': 1024, 'primary_arch': 'aarch64'}
Opening 0
current_mode 1280x720
MALI_CreateWindow:0x55a3653570 done.

**Comment by sergiogbrox** [comment]
https://github.com/user-attachments/assets/33580c49-146b-4276-8de7-0132f1b35e21

**Comment by cizia64** [comment]
Wow it crash after the launch. It's a particular case, it could be some corrupted files.
 [try reinstall portmaster](https://github.com/kloptops/TRIMUI_EX).

**Comment by sergiogbrox** [comment]
Yeah, it was probably something corrupted. I tried replacing all the files, deleting the system and leaving only the roms, imgs, bios and port masters and putting the system back, in short, I tried several things and nothing worked. The system only started working correctly after I backed up the roms, imgs, bios and port masters, formatted the memory card in slow mode and then transferred everything back to the card and reinstalled the system. It was over 200 GB of roms lol, it was a lot of work to do the backup, but it worked.

Thank you very much Cizia64 for your attention and patience in helping me, I appreciate it very much!"
